### ISSUE TITLE
Support writing MultiPoint into vector tiles

### ISSUE THREAD

**Issue by daniel-j-h** [systemed/tilemaker#719]
It looks like tilemaker supports points but only points and not multipoints.

Storing points in vector tiles is highly inefficient due to the heavy vector tile message encoding that's needed to store two numbers for a point. I found a historic ticket on this going into more details: https://github.com/mapbox/vector-tile-spec/issues/35

It would be great if tilemaker could write out MultiPoints where possible to
- make tiles smaller where many points are used, and
- to improve decoding and rendering efficiency in clients such as maplibre

---

For MultiPoints the vector tile geometry type is Point with multiple `MoveTo` commands.

Vector tile spec on MultiPoints
- https://github.com/mapbox/vector-tile-spec/tree/5330dfc6ba2d5f8c8278c2c4f56fff2c7dee1dbd/2.1#4342-point-geometry-type
- https://github.com/mapbox/vector-tile-spec/tree/5330dfc6ba2d5f8c8278c2c4f56fff2c7dee1dbd/2.1#4352-example-multi-point

The vector tile decoding library used e.g. in Maplibre
- https://github.com/mapbox/vector-tile-js/blob/77851380b63b07fd0af3d5a3f144cc86fb39fdd1/lib/vectortilefeature.js#L53-L71

MultiPoint read and write support in vtzero
- https://github.com/mapbox/vtzero/blob/67f934ee9aa44b923b5d0bc2bc87448900051a57/include/vtzero/geometry.hpp#L253-L278
- https://github.com/mapbox/vtzero/blob/67f934ee9aa44b923b5d0bc2bc87448900051a57/include/vtzero/geometry.hpp#L227-L251
- https://github.com/mapbox/vtzero/blob/67f934ee9aa44b923b5d0bc2bc87448900051a57/test/t/test_geometry_point.cpp#L59-L64

MultiPoint writing usage example in vtzero
- https://github.com/mapbox/vtzero/blob/67f934ee9aa44b923b5d0bc2bc87448900051a57/test/t/test_builder_point.cpp#L116-L154

Tilemaker writing out a point
- https://github.com/systemed/tilemaker/blob/9e6f7d58cb8634f762328af43b2382d148cef925/src/tile_worker.cpp#L278-L286

**Comment by systemed** [comment]
Good call. We [already sort features by attribute](https://github.com/systemed/tilemaker/blob/master/src/tile_data.cpp#L397-L424) so the trick here will be for the point-writing code (as you've identified above) to look ahead in the same way that the polygon and linestring writers just below already do, checking for `.compatible` and grouping if so.
